R/arc-select.R

```diff
@@ -111,18 +86,81 @@ arc_select <- function(
   )
 
   # return(x)
-  collect_layer(x, n_max = n_max)
+  collect_layer(x, n_max = n_max, error_call = error_call)
 }
 
+#' Apply filter_geom to x
+#'
+#' @keywords internal
+apply_filter_geom <- function(x,
+                              filter_geom = NULL,
+                              predicate = "intersects",
+                              crs = sf::st_crs(x),
+                              error_call = rlang::caller_env()) {
+  obj_check_arcgis(
+    filter_geom,
+    class = c("sfc", "sfg", "bbox"),
+    call = error_call
+    )
+
+  if (inherits(filter_geom, "bbox")) {
+    filter_geom <- sf::st_as_sfc(filter_geom)
+  }
+
+  if (inherits(filter_geom, "sfg")) {
+    filt_crs <- crs
+  }
 
+  # if its an sfc object it must be length one
+  if (inherits(filter_geom, "sfc")) {
+    if (length(filter_geom) > 1) {
+      filter_geom <- sf::st_union(filter_geom)
+    }
 
+    # if the CRS is missing use the layer's CRS
+    # otherwise use the CRS of the filter_geom object
+    filt_crs <- sf::st_crs(filter_geom)
+
+    if (is.na(filt_crs)) {
+      filt_crs <- crs
+    }
+
+    # extract the sfg object which is used to write Esri json
+    filter_geom <- filter_geom[[1]]
+  }
+
+  # if a multi polygon stop, must be a single polygon see
+  # related issue: https://github.com/R-ArcGIS/api-interface/issues/4
+  if (inherits(filter_geom, "MULTIPOLYGON")) {
+    cli::cli_warn(
+      c(
+        "{.arg filter_geom} can't have {.val MULTIPOLYGON} geometry.",
+        "v" = "Using {.fn sf::st_cast} to covert {.arg filter_geom} geometry to {.val POLYGON}"
```

Comment:
please use `"i" = ` since we are informing the user of a behavior rather than reporting a success

Reply:
I think we should inform the user that the multipolygon is being unioned before we tell them its being cast. Can we adjust the message to inform them that we're assuming the multipolygon is a coverage?

Reply:
I switched this to an cli_inform message since it isn't really a warning anymore. Unless you think this is really important, it may make sense to covert it into a `cli_bullets()` call because that can be suppressed with the cli default message handler option but cli_inform messages can only be suppressed by wrapping a function in `suppressMessages()`

Reply:
If it proves annoying, we can crib the `local_quiet_cli()` function I helped write for rextendr https://github.com/extendr/rextendr/blob/69eba4d1a7a9388ef708082a2204514f1621b773/R/utils.R#L26

Reply:
I'd need to double-check but I think that won't work if we use `cli_inform()`. That is why I'd suggest switching to `cli_bullets()` or `cli_alert()` for this message.